euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HBaseStreamer.java

```diff
@@ -0,0 +1,111 @@
+import cz.seznam.euphoria.hadoop.SerializableWritable;
+import cz.seznam.euphoria.shadow.com.google.common.base.Preconditions;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.KeyValue;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.hadoop.hbase.mapreduce.KeyValueSerialization.KeyValueSerializer;
+import org.apache.hadoop.io.serializer.Serializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Read and persist HBase table.
+ */
+public class HBaseStreamer implements Serializable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HBaseStreamer.class);
+
+  public static void main(String[] args) throws URISyntaxException, IOException {
+    Preconditions.checkArgument(
+        args.length == 3, "Please specify <input_uri> <output_uri> <executor>");
+
+    URI input = new URI(args[0]);
+    URI output = new URI(args[1]);
+    Executor executor = Executors.createExecutor(args[2]);
+    HBaseStreamer app = new HBaseStreamer(input, output, executor);
+    app.run();
+  }
+
+  private final URI input;
+  private final URI output;
+  private final SerializableWritable<Configuration> conf;
+  private final AtomicReference<Serializer<KeyValue>> serializer = new AtomicReference<>();
+  private final transient Executor executor;
+
+  private HBaseStreamer(URI input, URI output, Executor executor) {
+    this.input = input;
+    this.output = output;
+    this.executor = executor;
+    this.conf = new SerializableWritable<>(HBaseConfiguration.create());
+  }
+
+  private void run() {
+
+    Flow flow = Flow.create();
+    Dataset<Pair<ImmutableBytesWritable, Result>> ds = flow.createInput(
+        Utils.getHBaseSource(input, conf.get()));
+
+    FlatMap.of(ds)
+        .using((Pair<ImmutableBytesWritable, Result> p, Collector<byte[]> c) -> {
+          writeCellsAsBytes(p.getSecond(), c);
+        })
+        .output()
+        .persist(Utils.getSink(output, conf.get()));
+    LOG.info("Starting flow reading from {} and persisting to {}", input, output);
+    executor.submit(flow).join();
+  }
+
+  private void writeCellsAsBytes(Result res, Collector<byte[]> c) {
+
+    if (serializer.get() == null) {
+      serializer.set(new KeyValueSerializer());
+    }
+    final Serializer<KeyValue> s = serializer.get();
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    for (Cell cell : res.listCells()) {
+      try {
+        s.open(baos);
+        s.serialize((KeyValue) cell);
+        s.close();
+        c.collect(baos.toByteArray());
+        baos.reset();
+      } catch (IOException ex) {
```

Comment:
ExceptionUtils?

Reply:
Due to possible overhead I'll leave it unwrapped.